cmd/rpcdaemon/commands/eth_call.go

```diff
@@ -123,10 +124,13 @@ func (api *APIImpl) EstimateGas(ctx context.Context, argsOrNil *ethapi.CallArgs,
 		hi = uint64(*args.Gas)
 	} else {
 		// Retrieve the block to act as the gas ceiling
-		h, err := headerByNumberOrHash(dbtx, bNrOrHash, api)
+		h, err := headerByNumberOrHash(ctx, dbtx, bNrOrHash, api)
 		if err != nil {
 			return 0, err
 		}
+		if h == nil {
+			return 0, fmt.Errorf("no header found with number %d", *bNrOrHash.BlockNumber)
```

Comment:
I am still not sure about this - we have some RPC requests that require returning `nil` if the block is not found. Is this necessary?

Reply:
I believe we would actually get a nil ptr error when we try getting the gas limit from the header if we don't check it, since it is a possibility that we get nil as a response from `headerByBlockNumOrHash`

Reply:
He mean, most of API's doing next way:
```
if h == nil {
   return nil, nil
} 
```

Reply:
Ohh okay changed it, I misunderstood.